R/FilterState.R

```diff
@@ -336,42 +398,36 @@ InteractiveFilterState <- R6::R6Class( # nolint
     },
 
     #' @description
-    #' Set state.
+    #' Sets filtering state.
     #'
-    #' @param state (`list`)\cr
-    #'  contains fields relevant for a specific class:
-    #' \itemize{
-    #' \item{`selected`}{ defines initial selection}
-    #' \item{`keep_na` (`logical`)}{ defines whether to keep or remove `NA` values}
-    #' }
+    #' @param state a (`teal_slice`) object
     #'
-    #' @return NULL invisibly
+    #' @return `NULL` invisibly
     #'
     set_state = function(state) {
-      logger::log_trace(sprintf(
-        "%s$set_state, dataname: %s setting state of variable %s to: selected=%s, keep_na=%s",
-        class(self)[1],
-        private$dataname,
-        private$varname,
-        paste(state$selected, collapse = " "),
-        state$keep_na
-      ))
-      stopifnot(is.list(state) && all(names(state) %in% c("selected", "keep_na")))
+      checkmate::assert_class(state, "teal_slice")
```

Comment:
please add assert on `state$varname` to be the same as `private$varname`. Same for a `dataname`

Reply:
Adding assertion on `dataname` for higher levels as well.